src/TodoByTicketRule.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace staabm\PHPStanTodoBy;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use staabm\PHPStanTodoBy\utils\CommentMatcher;
+use staabm\PHPStanTodoBy\utils\ExpiredCommentErrorBuilder;
+use staabm\PHPStanTodoBy\utils\TicketStatusFetcher;
+
+use function in_array;
+use function strlen;
+use function trim;
+
+/**
+ * @implements Rule<Node>
+ */
+final class TodoByTicketRule implements Rule
+{
+    private const PATTERN = <<<'REGEXP'
+        /
+        @?TODO # possible @ prefix
+        @?[a-zA-Z0-9_-]*\s* # optional username
+        \s*[:-]?\s* # optional colon or hyphen
+        (?P<ticketKey>[A-Z0-9]+-\d+) # ticket key consisting of ABC-123 or F01-12345 format
+        \s*[:-]?\s* # optional colon or hyphen
+        (?P<comment>.*) # rest of line as comment text
+        /ix
+        REGEXP;
```

Comment:
since all previous parts are optional we should have a non-optional part to prevent false positives

```suggestion
        /
        @?TODO # possible @ prefix
        @?[a-zA-Z0-9_-]*\s* # optional username
        \s*[:-]?\s* # optional colon or hyphen
        \s+ # keyword/ticket separator
        (?P<ticketKey>[A-Z0-9]+-\d+) # ticket key consisting of ABC-123 or F01-12345 format
        \s*[:-]?\s* # optional colon or hyphen
        (?P<comment>.*) # rest of line as comment text
        /ix
        REGEXP;
```